### ISSUE TITLE
Infinite loop in HTML indentation when >1 buffers open

### ISSUE THREAD

**Issue by KeithScheiwiller** [neovim/neovim#8867]
- `nvim --version`:
```
NVIM v0.3.1
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wconversion -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/home/keith/Downloads/neovim-git/src/neovim-git/build/config -I/home/keith/Downloads/neovim-git/src/neovim-git/src -I/usr/include -I/home/keith/Downloads/neovim-git/src/neovim-git/build/src/nvim/auto -I/home/keith/Downloads/neovim-git/src/neovim-git/build/include
Compiled by keith@arch_w

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- Operating system/version:
`Arch Linux 4.17.11-arch1`
- Terminal name/version:
`urxvt v9.22`
- `$TERM`:
`rxvt-unicode-256color`

### Steps to reproduce using `nvim -u NORC`

`foo.html` & `foo2.html` (same contents):
```
<style lang="sass" src="foo.scss">
</style>

<template>
    <div>
    </div>
</template>

<script>
    import Vue from 'vue';

    const Foo = Vue.extend({
        components: {
        },

        events: {
        },

        data() {
        },

        computed: {
        },

        props: {
        },

        methods: {

        watch: {
        },

        ready() {
        },
    });

    export default Foo;
</script>
```

1) `nvim -u NORC foo.html`
2) `:e foo2.html`
3) Place cursor on `c` in `components:`
4) Type `o`

### Actual behaviour

neovim freezes without opening a new line. All input is ignored and CPU spikes up to 100%.
Pressing Ctrl-C will return control, but attempting to do an `o` again will return to locked state.

### Expected behaviour

neovim opens a new line in insert mode.

### Additional info

Note that ONLY occurs when >1 buffer is loaded. It doesn't seem to matter what the buffers are, so I used a copy of the same file for the repro.
My suspicions of it lying in `runtime/indent/html.vim` and `runtime/indent/javascript.vim` only comes from digging around in gdb while it's locked up. I unfortunately did not have the time to trace through it more thoroughly.

**Comment by chrisbra** [comment]
You might want to try, if this still happens, if you manually copy over the runtime files from Vim.

https://github.com/vim/vim/blob/master/runtime/indent/html.vim
https://github.com/vim/vim/blob/master/runtime/indent/javascript.vim

However I couldn't not reproduce the issue in a plain Vim with Neovims older indent files. (I don't have a neovim available).

**Comment by KeithScheiwiller** [comment]
The 2 indent files from vim's master do indeed fix the issue. Odd though that the issue only arose in the past month or so.
I'll try to bisect if I get the chance.

**Comment by janlazo** [comment]
Does this still happen on master? I want confirmation that it's a runtime issue, not because of recent fix from Vim `src/`.

**Comment by KeithScheiwiller** [comment]
@janlazo Yup, just built fresh from master using the default runtime and can still reproduce.

**Comment by henryhsue** [comment]
Able to reproduce on osx.

**Comment by xp-bar** [comment]
Heyo, I just came across this issue while using the latest build as well - I found that disabling `smartindent` fixed the issue for me, if that helps narrow your search.

I'm not sure what logs to post etc. so if you need anything let me know.

**Comment by justinmk** [comment]
We are up to date with Vim's `runtime/` as of Nvim 0.3.4 relative to when this issue was reported. So it should be fixed on Nvim 0.3.4.
